lldp_util.c

```diff
@@ -706,10 +704,23 @@ int is_macvtap(const char *ifname)
 		goto out_free;
 	}
 
-	memset(nlh, 0, NLMSG_SIZE);
+	do {
+		realsize = recv(s, NULL, 0, MSG_DONTWAIT | MSG_PEEK | MSG_TRUNC);
+	} while ((realsize < 0) && errno == EINTR);
+
+	if (realsize <= 0) {
+		goto out_free;
+	}
+
+	free(nlh);
+	nlh = calloc(1, realsize);
```

Comment:
A `realloc()` and `memset()` might be better here. For the 99% of cases, this will be shrinking the buffer size and `realloc()` avoids having to move it around.